aten/src/ATen/native/cuda/LegacyThrustHelpers.cu

```diff
@@ -30,4 +32,66 @@ void index_put_with_sort_kernel_thrust_helper(Tensor &linearIndex, Tensor &orig_
   thrust::sort_by_key(policy, sorted_data, sorted_data + num_indices, orig_data, LTOp<int64_t>());
 }
 
+template<typename index_t>
+int embedding_renorm_cuda_unique_copy(Tensor &indices_contig, Tensor &unique_indices) {
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+  auto allocator = THCThrustAllocator(globalContext().lazyInitCUDA());
+  auto policy = thrust::cuda::par(allocator).on(stream);
+
+  using device_ptr = thrust::device_ptr<index_t>;
+
+  auto num_indices = indices_contig.numel();
+  auto indices_data = device_ptr(indices_contig.data_ptr<index_t>());
+  auto unique_data = device_ptr(unique_indices.data_ptr<index_t>());
+  auto end = thrust::unique_copy(policy, indices_data, indices_data + num_indices, unique_data);
```

Comment:
unique_copy probably still queries device properties? Probably we can use cub DeviceSelect::Unique here?

Reply:
Yes, it does. And I will migrate this to cub, but will be in next PR :)